### PR TITLE
feat: Enabling DefaultAzureCredential for AuthenticationMode

### PR DESCRIPTION
**Pull request by jayendranarumugam**
Enabling `DefaultAzureCredential` for `AuthenticationMode` can support local debug/development operations and also can work on several Azure resources as well

With this improvement, we can now able to hit the default switch statement which earlier not possible 

```
            switch (authenticationInfo.Mode)
            {
                case AuthenticationMode.ServicePrincipal:
                    tokenCredential = new ClientSecretCredential(tenantId, authenticationInfo.IdentityId, authenticationInfo.Secret, tokenCredentialOptions);
                    break;
                case AuthenticationMode.UserAssignedManagedIdentity:
                    var clientId = authenticationInfo.GetIdentityIdOrDefault();
                    tokenCredential = new ManagedIdentityCredential(clientId, tokenCredentialOptions);
                    break;
                case AuthenticationMode.SystemAssignedManagedIdentity:
                    tokenCredential = new ManagedIdentityCredential(options:tokenCredentialOptions);
                    break;
                default:
                    tokenCredential = new DefaultAzureCredential();   // <-- this can possible now
                    break;
            }
```

I successfully tested this from my local and validated the Unit Test cases. Please let me know your thoughts on this feature

### CODE REVIEW

File: changelog/content/deprecated/authentication-modes.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ However, as of Promitor Scraper v2.2.0 & Resource Discovery v0.3.0, users can ch
 
 ```yaml
 authentication:
-  # Options are ServicePrincipal, SystemAssignedManagedIdentity, UserAssignedManagedIdentity.
+  # Options are ServicePrincipal, SystemAssignedManagedIdentity, UserAssignedManagedIdentity , DefaultAzureCredential.
```
Comment (tomkerkhove) [inline changelog/content/deprecated/authentication-modes.md:25]: Let's rename it to decouple from SDK
```suggestion
  # Options are ServicePrincipal, SystemAssignedManagedIdentity, UserAssignedManagedIdentity , SdkDefault.
```

**Review by tomkerkhove — CHANGES_REQUESTED** [review]
Thanks for the configuration, 1 suggestion.

Can you also please open a PR for the docs?

File: changelog/content/deprecated/authentication-modes.md (hunk 1 above)
Comment (jayendranarumugam) [reply changelog/content/deprecated/authentication-modes.md]: Renamed as per the suggestion

**Comment by jayendranarumugam** [comment]
Opened a PR https://github.com/promitor/docs/pull/67/files for relevant docs update

**Review by tomkerkhove — APPROVED** [review]

**Comment by tomkerkhove** [comment]
@jayendranarumugam Mind fixing the code quality?

**Comment by jayendranarumugam** [comment]
@tomkerkhove, sure, this is relatively new to me. Please correct my understanding; we have to fix all the 

` ##[notice]"[]......` to make this code quality to pass ?

**Comment by tomkerkhove** [comment]
@jayendranarumugam Looks like we are not there yet, mind fixing build issue please?

**Comment by jayendranarumugam** [comment]
@tomkerkhove pushed updates could you trigger the code scan and CI workflow now ?

**Comment by jayendranarumugam** [comment]
@tomkerkhove checked the latest failed logs on code quality which seems confusing me

Earlier the [code quality failed with ](https://github.com/tomkerkhove/promitor/actions/runs/12007318471/job/33468659849)
> ##[warning]"[InconsistentNaming] Name 'azureMetricConfigurationBase' does not match rule 'Static readonly fields (private)'. Suggested name is 'AzureMetricConfigurationBase'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Metrics/ScrapeDefinitionBatchPropertiesTest.cs(22,60)

So after i renamed the variable as per the suggestion i.e `AzureMetricConfigurationBase`

But now [i got the error to undo it the old way](https://github.com/tomkerkhove/promitor/actions/runs/12117711249/job/33809090751)

> ##[warning]"[InconsistentNaming] Name 'AzureMetricConfigurationBase' does not match rule 'Static readonly fields (private)'. Suggested name is 'azureMetricConfigurationBase'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Metrics/ScrapeDefinitionBatchPropertiesTest.cs(22,60)

**Comment by tomkerkhove** [comment]
I believe these are the ones to tackle, no?

```
Warning: "[InconsistentNaming] Name 'AzureMetricConfigurationBase' does not match rule 'Static readonly fields (private)'. Suggested name is 'azureMetricConfigurationBase'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Metrics/ScrapeDefinitionBatchPropertiesTest.cs(22,60)
Warning: "[InconsistentNaming] Name 'AzureMetricConfigurationBase' does not match rule 'Static readonly fields (private)'. Suggested name is 'azureMetricConfigurationBase'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Scraping/Batching/AzureResourceDefinitionBatchingTests.cs(24,60)
Warning: "[InconsistentNaming] Name 'AzureMetricNameBase' does not match rule 'Static readonly fields (private)'. Suggested name is 'azureMetricNameBase'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Metrics/ScrapeDefinitionBatchPropertiesTest.cs(18,40)
Warning: "[InconsistentNaming] Name 'AzureMetricNameBase' does not match rule 'Static readonly fields (private)'. Suggested name is 'azureMetricNameBase'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Scraping/Batching/AzureResourceDefinitionBatchingTests.cs(20,40)
Warning: "[InconsistentNaming] Name 'AzureMonitorMeter' does not match rule 'Static readonly fields (private)'. Suggested name is 'azureMonitorMeter'." on /home/runner/work/promitor/promitor/src/Promitor.Integrations.Sinks.OpenTelemetry/OpenTelemetryCollectorMetricSink.cs(20,39)
Warning: "[InconsistentNaming] Name 'BatchSize' does not match rule 'Static readonly fields (private)'. Suggested name is 'batchSize'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Scraping/Batching/AzureResourceDefinitionBatchingTests.cs(45,37)
Warning: "[InconsistentNaming] Name 'Bogus' does not match rule 'Static readonly fields (private)'. Suggested name is 'bogus'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Generators/ScrapeResultGenerator.cs(10,39)
Warning: "[InconsistentNaming] Name 'DefaultAggregationInterval' does not match rule 'Static readonly fields (private)'. Suggested name is 'defaultAggregationInterval'." on /home/runner/work/promitor/promitor/src/Promitor.Core.Scraping/Configuration/Serialization/v1/Core/AggregationDeserializer.cs(9,42)
Warning: "[InconsistentNaming] Name 'DefaultInterval' does not match rule 'Static readonly fields (private)'. Suggested name is 'defaultInterval'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Serialization/DeserializerTests/DeserializationTests.cs(13,42)
Warning: "[InconsistentNaming] Name 'EmptyResourceDefinitions' does not match rule 'Static readonly fields (private)'. Suggested name is 'emptyResourceDefinitions'." on /home/runner/work/promitor/promitor/src/Promitor.Agents.Scraper/Discovery/StubResourceDiscoveryRepository.cs(11,63)
Warning: "[InconsistentNaming] Name 'HealthReport' does not match rule 'Static readonly fields (private)'. Suggested name is 'healthReport'." on /home/runner/work/promitor/promitor/src/Promitor.Agents.Scraper/Discovery/StubResourceDiscoveryRepository.cs(12,51)
Warning: "[InconsistentNaming] Name 'LogAnalyticsConfigurationBase' does not match rule 'Static readonly fields (private)'. Suggested name is 'logAnalyticsConfigurationBase'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Metrics/ScrapeDefinitionBatchPropertiesTest.cs(30,61)
Warning: "[InconsistentNaming] Name 'LogAnalyticsConfigurationBase' does not match rule 'Static readonly fields (private)'. Suggested name is 'logAnalyticsConfigurationBase'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Scraping/Batching/AzureResourceDefinitionBatchingTests.cs(32,61)
Warning: "[InconsistentNaming] Name 'NamingConvention' does not match rule 'Static readonly fields (private)'. Suggested name is 'namingConvention'." on /home/runner/work/promitor/promitor/src/Promitor.Core/Serialization/Yaml/YamlSerialization.cs(8,51)
Warning: "[InconsistentNaming] Name 'PrometheusMetricDefinition' does not match rule 'Static readonly fields (private)'. Suggested name is 'prometheusMetricDefinition'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Metrics/ScrapeDefinitionBatchPropertiesTest.cs(19,60)
Warning: "[InconsistentNaming] Name 'PrometheusMetricDefinitionTest' does not match rule 'Static readonly fields (private)'. Suggested name is 'prometheusMetricDefinitionTest'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Scraping/Batching/AzureResourceDefinitionBatchingTests.cs(21,60)
Warning: "[InconsistentNaming] Name 'ResourceGroupNameTest' does not match rule 'Static readonly fields (private)'. Suggested name is 'resourceGroupNameTest'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Scraping/Batching/AzureResourceDefinitionBatchingTests.cs(44,40)
Warning: "[InconsistentNaming] Name 'ScrapingBase' does not match rule 'Static readonly fields (private)'. Suggested name is 'scrapingBase'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Metrics/ScrapeDefinitionBatchPropertiesTest.cs(38,44)
Warning: "[InconsistentNaming] Name 'ScrapingBase' does not match rule 'Static readonly fields (private)'. Suggested name is 'scrapingBase'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Scraping/Batching/AzureResourceDefinitionBatchingTests.cs(40,44)
Warning: "[InconsistentNaming] Name 'SubscriptionId' does not match rule 'Static readonly fields (private)'. Suggested name is 'subscriptionId'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Metrics/ScrapeDefinitionBatchPropertiesTest.cs(21,40)
Warning: "[InconsistentNaming] Name 'SubscriptionIdTest' does not match rule 'Static readonly fields (private)'. Suggested name is 'subscriptionIdTest'." on /home/runner/work/promitor/promitor/src/Promitor.Tests.Unit/Core/Scraping/Batching/AzureResourceDefinitionBatchingTests.cs(23,40)
```

File: src/Promitor.Agents.Scraper/Discovery/StubResourceDiscoveryRepository.cs [hunk 2]
```diff
@@ -8,17 +8,17 @@ namespace Promitor.Agents.Scraper.Discovery
 {
     public class StubResourceDiscoveryRepository : IResourceDiscoveryRepository
     {
-        private static readonly List<AzureResourceDefinition> emptyResourceDefinitions = new();
-        private static readonly AgentHealthReport healthReport = new();
+        private static readonly List<AzureResourceDefinition> EmptyResourceDefinitions = new();
```
Comment (tomkerkhove) [inline src/Promitor.Agents.Scraper/Discovery/StubResourceDiscoveryRepository.cs:11]: Let's just revert to the old name if this is not related to your change, which applies to all these

File: src/Promitor.Integrations.AzureMonitor/HttpPipelinePolicies/RecordArmRateLimitMetricsPolicy.cs [hunk 3]
```diff
@@ -74,6 +74,8 @@ private string DetermineApplicationId(AzureAuthenticationInfo azureAuthenticatio
                     return azureAuthenticationInfo.GetIdentityIdOrDefault("externally-configured-user-assigned-identity");
                 case AuthenticationMode.SystemAssignedManagedIdentity:
                     return "system-assigned-identity";
+                case AuthenticationMode.SdkDefault:
+                    return "default-azure-credentials";
```
Comment (tomkerkhove) [inline src/Promitor.Integrations.AzureMonitor/HttpPipelinePolicies/RecordArmRateLimitMetricsPolicy.cs:78]:
```suggestion
                    return "default-azure-credentials";
```

File: src/Promitor.Integrations.AzureMonitor/RequestHandlers/AzureResourceManagerThrottlingRequestHandler.cs [hunk 4]
```diff
@@ -97,6 +97,8 @@ private string DetermineApplicationId(AzureAuthenticationInfo azureAuthenticatio
                     return azureAuthenticationInfo.GetIdentityIdOrDefault("externally-configured-user-assigned-identity");
                 case AuthenticationMode.SystemAssignedManagedIdentity:
                     return "system-assigned-identity";
+                case AuthenticationMode.SdkDefault:
+                    return "default-azure-credentials";
```
Comment (tomkerkhove) [inline src/Promitor.Integrations.AzureMonitor/RequestHandlers/AzureResourceManagerThrottlingRequestHandler.cs:101]:
```suggestion
                    return "default-azure-credentials";
```

File: src/Promitor.Tests.Unit/Validation/Authentication/AzureAuthenticationValidationStepTests.cs [hunk 5]
```diff
@@ -304,6 +304,26 @@ public void SystemAssignedManagedIdentity_ValidWithoutApplicationKey_Succeeds()
             PromitorAssert.ValidationIsSuccessful(validationResult);
         }
 
+
```
Comment (tomkerkhove) [inline src/Promitor.Tests.Unit/Validation/Authentication/AzureAuthenticationValidationStepTests.cs:307]:
```suggestion
```

Comment (jayendranarumugam) [reply src/Promitor.Tests.Unit/Validation/Authentication/AzureAuthenticationValidationStepTests.cs]: updated

File: src/Promitor.Integrations.AzureMonitor/HttpPipelinePolicies/RecordArmRateLimitMetricsPolicy.cs (hunk 3 above)
Comment (jayendranarumugam) [reply src/Promitor.Integrations.AzureMonitor/HttpPipelinePolicies/RecordArmRateLimitMetricsPolicy.cs]: updated

File: src/Promitor.Agents.Scraper/Discovery/StubResourceDiscoveryRepository.cs (hunk 2 above)
Comment (jayendranarumugam) [reply src/Promitor.Agents.Scraper/Discovery/StubResourceDiscoveryRepository.cs]: reverted

**Review by tomkerkhove — APPROVED** [review]

**Comment by tomkerkhove** [comment]
Thanks a ton!

**Comment by tomkerkhove** [comment]
@jayendranarumugam Did you manage to open a doc PR?

**Comment by jayendranarumugam** [comment]
@tomkerkhove please check https://github.com/promitor/docs/pull/67
